Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -469,6 +469,143 @@ def template_test(template: str):
+    timeout = int(args.get('timeout', 600))
+
+    ScheduledCommand.raise_error_if_not_supported()
+
+    # Create a new job if there is no job_id
+    if not job_id:
+
+        result = search_function()
+
+        # If the job was started successfully
+        if 'result' in result.get('response', None) and result.get('response', {}).get('result', {}).get('job', None):
+
+            job_id = result.get('response', {}).get('result', {}).get('job', None)
+            output = {
+                'JobID': job_id,
+                'Status': 'Pending',
+                'Description': args.get('description')
+            }
+            if additional_paths:
+                for k, v in additional_paths.items():
+                    output[k] = v
+
+            polling_args = {
+                'job_id': job_id,
+                'polling': True,
+                **args
+            }
+
+            scheduled_command = ScheduledCommand(
+                command=cmd,
+                next_run_in_seconds=interval_in_seconds,
+                args=polling_args,
+                timeout_in_seconds=timeout
+            )
+            readable_output = f"Running {cmd} (Job ID: {job_id})"
+
+            polling_results.append(CommandResults(
+                outputs_prefix=outputs_prefix,
+                outputs_key_field='JobID',
+                outputs=output,
+                readable_output=readable_output,
+                scheduled_command=scheduled_command,
+                ignore_auto_extract=True
+            ))
+
+        # Otherwise return the error information
+        else:
+            if error_path:
+                error = result
+                for key in error_path:
+                    error = error.get(key, {})
+            else:
+                error = 'Unknown'
+            polling_results.append(f"{exit_string} - {error}")
+
+    # Check existing job
+    else:
+
+        result = query_function()
+
+        # Reschedule job if it's not complete
+        if result['response']['result']['job']['status'] != 'FIN':
+            polling_args = {
+                'job_id': job_id,
+                'polling': True,
+                **args
+            }
+
+            scheduled_command = ScheduledCommand(
+                command=cmd,
+                next_run_in_seconds=interval_in_seconds,
+                args=polling_args,
+                timeout_in_seconds=timeout
+            )
+            polling_results.append(
+                CommandResults(
+                    scheduled_command=scheduled_command
+                )
+            )
+
+        # Check the status of a completed job
+        else:
+            status_output = {'JobID': job_id}
+            if additional_paths:
+                for k, v in additional_paths.items():
+                    status_output[k] = v
+
+            status = result
+            for key in success_path[0]:
+                status = status.get(key, {})
```

Comment:
I've looked at the function and whilst it is a cool function, the requirement here is not to retrun a default value (or rather only return a blank dictionary) which the code above does in 2 lines. Otherwise the reader would then have to refer to the "CommonServerPython" for a function that is then carried out in 12 lines. For readability and simplicity I think the above solution works better?

What are your thoughts?